go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -221,6 +173,81 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 			errs.RecordError(err)
 		}
 	}()
+	wg.Wait()
+
+	// TODO(deepthi) if there was a topo.NoUpdateNeeded error then we should probably not do the following?
```

Comment:
Historically we've seen that the shard record is often out of sync with the tablet records, because there are no transactions. The state of one doesn't necessarily imply we don't still need to update the other.